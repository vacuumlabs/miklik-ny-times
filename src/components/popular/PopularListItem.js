import Avatar from '@material-ui/core/Avatar'
import ListItem from '@material-ui/core/ListItem'
import ListItemAvatar from '@material-ui/core/ListItemAvatar'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import ListItemText from '@material-ui/core/ListItemText'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import RightIcon from '@material-ui/icons/ChevronRight'
import DateIcon from '@material-ui/icons/DateRange'
import React from 'react'
import { ItemPropType } from '../../app/popular/popularTypes';

const useStyles = makeStyles(theme => ({
  secondary: {
    display: 'flex',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
  },
  date: {
    display: 'flex',
    alignItems: 'center',
  },
}))

export default function PopularListItem({ item }) {
  const classes = useStyles()

  return (
    <ListItem button>
      <ListItemAvatar>
        <Avatar alt="Thumbnail" src={item.thumbnail} />
      </ListItemAvatar>
      <ListItemText
        primary={item.title}
        secondary={
          <span className={classes.secondary}>
            <Typography component="span" variant="body2" color="textPrimary">
              {item.authors}
            </Typography>
            <span className={classes.date}>
              <DateIcon fontSize="small" />
              {item.date}
            </span>
          </span>
        }
      />
      <ListItemSecondaryAction>
        <RightIcon />
      </ListItemSecondaryAction>
    </ListItem>
  )
}

PopularListItem.propTypes = {
    item: ItemPropType.isRequired,
}