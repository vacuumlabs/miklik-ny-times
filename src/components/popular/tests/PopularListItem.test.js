import React from 'react'
import { render } from '@testing-library/react'
import PopularListItem from '../PopularListItem'
import { firstArticle } from '../../../tests/articleFixture'
import { commonSetup } from '../../../tests/testUtils'

describe('PopularListItem', () => {
  commonSetup()

  it('prints one list item', () => {
    const { container } = render(
      <PopularListItem item={firstArticle} onSelect={() => 0} />
    )
    expect(container).toMatchSnapshot()
  })
})
