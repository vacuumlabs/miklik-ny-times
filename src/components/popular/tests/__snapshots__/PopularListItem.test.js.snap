// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularListItem prints one list item 1`] = `
<div>
  <li
    class="MuiListItem-container"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root"
        >
          <img
            alt="Thumbnail"
            class="MuiAvatar-img"
            src="https://static01.nyt.com/images/2018/06/04/us/05xp-graduation2/05xp-graduation2-thumbStandard.jpg"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          Kentucky Crowd Cheers Valedictorianâ€™s Trump Quote, Then Learns Obama Said It
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          <span
            class="makeStyles-secondary-1"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              By LAURA M. HOLSON
            </span>
            <span
              class="makeStyles-date-2"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-dateIcon-3 MuiSvgIcon-fontSizeSmall"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              2018-06-04
            </span>
          </span>
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiListItemSecondaryAction-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </li>
</div>
`;
